src/main/java/mate/academy/controller/BookingController.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@Tag(name = "Booking management", description = "Endpoints for managing booking")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/bookings")
+public class BookingController {
+    private final BookingService bookingService;
+    private final UserService userService;
+
+    @PreAuthorize("hasAuthority('ROLE_USER') or hasAuthority('ROLE_ADMIN')")
+    @GetMapping("/my")
+    @Operation(summary = "Get all bookings", description = "Get a list of all available bookings")
+    public List<BookingDto> getAllBookings(Authentication authentication,
+                                           Pageable pageable,
+                                           @RequestParam(value = "userId",
+                                                   required = false) Long userId) {
+        boolean isAdmin = authentication.getAuthorities()
+                .contains(new SimpleGrantedAuthority("ROLE_ADMIN"));
+
+        if (isAdmin && userId != null) {
+            return bookingService.getBookingsByUserId(userId, pageable);
+        } else {
+            return bookingService.getAllBookingsWithoutUserId(pageable);
+        }
+    }
+
+    @GetMapping("/{id}")
+    public BookingDto getBookingById(@PathVariable Long id) {
+        return bookingService.getBookingById(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public BookingDto createBooking(@RequestBody @Valid CreateBookingRequestDto request,
+                                    Authentication authentication) {
+        UserDetails principal = (UserDetails) authentication.getPrincipal();
+        String email = principal.getUsername();
+
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
```

Comment:
use custom exception 
provide userId to message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("User not found"));
```